src/client/src/apps/StyleguideRoute/components/OnePageNavBar.tsx

```diff
@@ -20,49 +30,95 @@ const OnePageNavBar: React.FC<OnePageNavBar> = props => {
   const [scrollTop, setScrollTop] = useState(window.scrollY)
   const [positionNavBar, setPositionNavBar] = useState(MAX_SCROLL_HEIGHT)
   const [currentSection, setCurrentSection] = useState('')
-  const handleScroll = () => setScrollTop(window.scrollY)
+  const [isSticky, setIsSticky] = useState(false)
+
+  const getCurrentSection = useCallback((): NavSection => {
+    let currentSection = undefined
+    const currentScroll = window.scrollY
+
+    sections.forEach((section: NavSection) => {
+      if (section) {
+        if (section.ref) {
+          if (section.ref.current) {
+            const borderTop = section.ref.current.offsetTop - DEFAULT_OFFSET
+            const borderBottom =
+              section.ref.current.offsetTop + section.ref.current.offsetHeight - DEFAULT_OFFSET
+
+            if (currentScroll >= borderTop && currentScroll < borderBottom) {
+              currentSection = section
+            }
+          }
+        }
+      }
+    })
+
+    return currentSection
+  }, [sections])
+
+  const handleScroll = useCallback(() => {
+    setScrollTop(window.scrollY)
+    const currentSectionScrolled: NavSection = getCurrentSection()
+
+    if (typeof currentSectionScrolled !== 'undefined') {
```

Comment:
Can we add another conditional check in here. For example:

```javascript
&& !location.hash.includes(currentSectionScrolled.path)
```

or 

```javascript
&& currentSectionScrolled.title !== currentSection
```

I think it will significantly improve performance as calling `history.pushState` is relatively slow; if we can only call it when needed, that will help :-)